fields_test.go

```diff
@@ -0,0 +1,263 @@
+package environschema_test
+
+import (
+	"github.com/juju/schema"
+	"gopkg.in/juju/environschema.v1"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+type suite struct{}
+
+var _ = gc.Suite(&suite{})
+
+type valueTest struct {
+	about       string
+	val         interface{}
+	expectError string
+	expectVal   interface{}
+}
+
+var validationSchemaTests = []struct {
+	about       string
+	fields      environschema.Fields
+	expectError string
+	tests       []valueTest
+}{{
+	about: "regular schema",
+	fields: environschema.Fields{
+		"stringvalue": {
+			Type: environschema.Tstring,
+		},
+		"mandatory-stringvalue": {
+			Type:      environschema.Tstring,
+			Mandatory: true,
+		},
+		"intvalue": {
+			Type: environschema.Tint,
```

Comment:
What about testing for secret? Maybe a bit too much but can bool/int be mandatory?